### PR TITLE
create django data model for contributors backend data

### PR DESCRIPTION
**Pull request by yashraj-Dhanuka**
Closes https://github.com/coala/community/issues/32

### CODE REVIEW

File: data/models.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    def contributors_data_to_json(self, repo_data):
+        self.contributors_data = json.dumps(repo_data)
+
+    def contributors_data_from_json(self):
+        return json.loads(self.contributors_data)
+
+
+class Contributor(models.Model):
+    login = models.TextField(default=None, primary_key=True)
```
Comment (ashwini0529) [inline data/models.py]: What is the use of login as primary key?

File: data/models.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    name = models.TextField(default=None, null=True)
+    bio = models.TextField(default=None, null=True)
+    num_commits = models.IntegerField(default=None, null=True)
```
Comment (ashwini0529) [inline data/models.py]: shouldn't it be `default=0`?

File: data/cron.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        c.name = data[contributor]['name']
+        c.bio = data[contributor]['bio']
+        c.num_commits = data[contributor]['contributions']
+        c.issues_opened = data[contributor]['issues']
+        c.reviews = data[contributor]['reviews']
+        c.save()
+
+
+def get_commits(repo, data):
+    url = '/repos/coala/' + repo['name'] + '/contributors'
```
Comment (ashwini0529) [inline data/cron.py]: you can use `repo_name` from `git.py` implemented in https://github.com/coala/community/pull/48

File: data/cron.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    response = GitHub.get(GITHUB_API_KEY, url)
+    for contributor in response:
+        data[contributor['login']]['contributions'] +=\
+            contributor['contributions']
+    return data
+
+
+def get_issues(repo, data):
+    url = '/repos/coala/' + repo['name'] + '/issues'
```
[the remark this answers is not in the document]
Comment (ashwini0529) [reply data/cron.py]: same as above

**Comment by ashwini0529** [comment]
It's still a lot to do. Please refer to @jayvdb's comment.

File: data/models.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    def contributors_data_to_json(self, repo_data):
+        self.contributors_data = json.dumps(repo_data)
+
+    def contributors_data_from_json(self):
+        return json.loads(self.contributors_data)
+
+
+class Contributor(models.Model):
+    login = models.TextField(default=None, primary_key=True)
+    name = models.TextField(default=None, null=True)
```
Comment (Monal5031) [inline data/models.py]: `name` should be a `CharField` IMO.

File: data/cron.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        data = get_issues(repo, commit_data)
+
+    for contributor in data:
+        response = \
+            requests.get('https://api.github.com/search/issues?q=commenter:' +
+                         contributor + '%20is:pr%20user:coala',
+                         headers=request_headers, timeout=3000)
+        r_json = json.loads(response.text)
+        data[contributor]['reviews'] = r_json['total_count']
+        if int(response.headers['X-RateLimit-Remaining']) == 1:
```
Comment (Monal5031) [inline data/cron.py]: Can't we simply keep it string comparison?

File: data/cron.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
+import json
+from collections import defaultdict
+from time import sleep
+
+import requests
+from IGitt.GitHub import GitHub
+
+from coala_web.settings import GITHUB_API_KEY
```
[the remark this answers is not in the document]
Comment (jayvdb) [reply data/cron.py]: This new app does not use GitHub.  It uses the Web service provided by http://gitlab.com/coala/landing 

Please read the issue more carefully, especially https://github.com/coala/community/issues/32#issuecomment-352238381

if you do not radically improve this PR the next time you push it, I will close this PR and unassign you from the issue.

File: data/cron.py (hunk 6 above)
Comment (jayvdb) [reply data/cron.py]: fyi, this is irrelevant; the PR isnt doing what the issue requests.  This entire module must be removed.

File: data/cron.py (hunk 3 above)
Comment (jayvdb) [reply data/cron.py]: irrelevant; the entire module `cron.py` must be deleted as it is nothing like what is requested.

**Comment by jayvdb** [comment]
No progress, and hasnt been going in the right direction.

**Pull request closed without merging**